Symfony/CS/Fixer/Contrib/EregToPregFixer.php

```diff
@@ -0,0 +1,173 @@
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Matteo Beccati <matteo@beccati.com>
+ */
+class EregToPregFixer extends AbstractFixer
+{
+    /**
+     * @var array the list of the ext/ereg function names, their preg equivalent and the preg modifier(s), if any
+     *            all condensed in an array of arrays.
+     */
+    private static $functions = array(
+        array('ereg', 'preg_match', ''),
+        array('eregi', 'preg_match', 'i'),
+        array('ereg_replace', 'preg_replace', ''),
+        array('eregi_replace', 'preg_replace', 'i'),
+        array('split', 'preg_split', ''),
+        array('spliti', 'preg_split', 'i'),
+    );
+
+    /**
+     * @var array the list of preg delimiters, in order of preference.
+     */
+    private static $delimiters = array('/', '#', '!');
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        if (!$this->cursoryMatch($content)) {
+            return $content;
+        }
+
+        $tokens = Tokens::fromCode($content);
+
+        foreach (self::$functions as $map) {
+            // the sequence is the function name, followed by "(" and a quoted string
+            $seq = array(array(T_STRING, $map[0]), '(', array(T_CONSTANT_ENCAPSED_STRING));
+
+            $currIndex = 0;
+            while (null !== $currIndex) {
+                $ereg = $tokens->findSequence($seq, $currIndex, $tokens->count(), false);
+
+                // did we find a match?
+                if (null === $ereg) {
+                    break;
+                }
+
+                // findSequence also returns the tokens, but we're only interested in the indexes, i.e.:
+                // 0 => function name,
+                // 1 => bracket "("
+                // 2 => quoted string passed as 1st parameter
+                $ereg = array_keys($ereg);
+
+                // advance tokenizer cursor
+                $currIndex = $ereg[2];
+
+                // ensure it's a function call (not a method / static call)
+                $prev = $tokens->getPrevMeaningfulToken($ereg[0]);
+                if (null === $prev || $tokens[$prev]->isGivenKind(array(T_OBJECT_OPERATOR, T_DOUBLE_COLON))) {
+                    continue;
+                }
+
+                // ensure the first parameter is just a string (e.g. has nothing appended)
+                $next = $tokens->getNextMeaningfulToken($ereg[2]);
+                if (null === $next || !$tokens[$next]->equalsAny(array(',', ')'))) {
+                    continue;
+                }
+
+                // convert to PCRE
+                $string = substr($tokens[$ereg[2]]->getContent(), 1, -1);
+                $quote = substr($tokens[$ereg[2]]->getContent(), 0, 1);
+                $delim = $this->getBestDelimiter($string);
+                $preg = $delim.addcslashes($string, $delim).$delim.'D'.$map[2];
```

Comment:
Indeed. It is unlikely that code using ereg is also checking the result with "===", but nevertheless the warning is well deserved. I'm also not 100% sure that any converted ereg will work exactly the same way as before, so some caution is advisable.